runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1542,3 +1554,166 @@ TR_J9SharedCacheServerVM::getInstanceFieldOffset(TR_OpaqueClassBlock * classPoin
 
    return ~0;
    }
+
+TR_OpaqueClassBlock *
+TR_J9SharedCacheServerVM::getSuperClass(TR_OpaqueClassBlock * classPointer)
+   {
+   TR::Compilation* comp = _compInfo->getCompInfoForCompOnAppThread() ? _compInfo->getCompInfoForCompOnAppThread()->getCompilation() : _compInfoPT->getCompilation();
```

Comment:
The JITaaS server does not support compilation on application thread (there are no application threads at the server). Thus, we can simplify the code to: `TR::Compilation* comp = _compInfoPT->getCompilation()`
Similar observation for other methods below.